universe/vault/credentials.go

```diff
@@ -8,11 +8,17 @@ import (
 	"context"
 	"encoding/json"
 	"os"
+	"sync"
+	"time"
 
 	"github.com/hashicorp/vault-client-go"
 	"github.com/hashicorp/vault-client-go/schema"
+	"github.com/rs/zerolog"
 )
 
+// Force re-authentication if the token expires in less than this much time.
+const ttlBuffer = time.Second * 8
```

Comment:
sounds very tight.
What are typical TTLs?
How about `time.Minute` as a buffer?

Reply:
The default is an hour, so a minute should be fine as well. I don't expect time skew of anything over a second, but it doesn't hurt much to be a bit more conservative.